content/blog/asyncapi-github-actions.md

```diff
@@ -0,0 +1,154 @@
+- We also created [official AsyncAPI action](https://github.com/marketplace/actions/generator-for-asyncapi-documents) for our [generator](https://github.com/asyncapi/generator/).
+
+## Writing a GitHub Action
+
+Our actions are both [written in JavaScript](https://help.github.com/en/actions/building-actions/creating-a-javascript-action). The other way of writing action is to do a [Docker container action](https://help.github.com/en/actions/building-actions/creating-a-docker-container-action). The best way to start writing your action is to:
+
+1. Follow [this](https://help.github.com/en/actions/building-actions/creating-a-javascript-action) tutorial to create a simple action to understand it's components.
+1. Get familiar with [official toolkit](https://github.com/actions/toolkit) that you can use to simplify writing an action. 
+1. Create your custom action with [this](https://github.com/actions/javascript-action) template that has many things plugged in already, like eslint, testing, and most important, distro generation, so you do not have to commit `node_modules` directory to your repository.
+
+These are all the resources I used to write my first action, and to master it, I only had to read official docs, like [reference docs for "action.yml" file](https://help.github.com/en/actions/building-actions/metadata-syntax-for-github-actions). Well done GitHub!
+
+## What I can do today with AsyncAPI GitHub Actions
+
+Those two actions can help you a lot already, together or separately. I present in this post only two possible workflows, and you can take it from here and think about your ideas.
+
+### Validation of AsyncAPI files in a Pull Request
+
+You can make sure that whenever someone makes a Pull Request to propose a change in the AsyncAPI document, you can validate it automatically using [Waleed's](https://twitter.com/WaleedAshraf01/) action `WaleedAshraf/asyncapi-github-action@v0.0.3`.
+
+Actions can be triggered by [multiple types of events](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions). In this example, we will trigger the action on any `pull_request` event.
+
+```yaml
+name: Validate AsyncAPI document
+
+on:
+  pull_request:
+  
+jobs:
+  validation:
+    runs-on: ubuntu-latest
+    - name: asyncapi-github-action
+      uses: WaleedAshraf/asyncapi-github-action@v0.0.3
+      with:
+        filepath: 'my-directory/asyncapi.yaml'
+```
+
+### Generating HTML and publishing it to GitHub Pages
+
+One of the AsyncAPI use cases is to document your application with it and generated docs out of it, best in HTML. The typical workflow here would be to have a GitHub Action workflow that your trigger on every push to the `master` branch. 
+
+```yaml
+name: AsyncAPI documentation publishing
+
+on:
+  push:
+    branches: [ master ]
+```
+
+To generate HTML from your AsyncAPI documentation, you need to use `asyncapi/github-action-for-generator@v0.2.0` action. You also need to specify a few more things:
```

Comment:
```suggestion
To generate HTML from your AsyncAPI definition, you need to use `asyncapi/github-action-for-generator@v0.2.0` action. You also need to specify a few more things:
```

Try not to call AsyncAPI documentation. Use either document or definition. Otherwise, people could think AsyncAPI is just for documentation. This is a problem OpenAPI had since the beginning because Swagger was completely focused on documentation only.

Reply:
I like definition.
@derberg and I also had long conv on this. I always called it AsyncAPI Schema.

Yes, there is a real AsyncAPI Schema also, but, to me, when someone defines their own AsyncAPI yaml file, it's also a schema?

But, calling it **definition** is much better.

Reply:
Yes, the AsyncAPI file is a schema but the hyper-schema that we use to validate it is also a schema, and the schemas inside the file are also schemas, etc. It's so generic that it's like saying that it's a file/document but "file" and "document" are less abused in our context.